go.mod

```diff
@@ -1,11 +1,18 @@
 module github.com/ovh/configstore
 
-go 1.12
+go 1.21.0
+
+toolchain go1.24.4
+
+require (
+	github.com/fsnotify/fsnotify v1.9.0
+	github.com/goccy/go-yaml v1.18.0
```

Comment:
I do not see any reference in the documentation of this library successfully parsing JSON payload
Does it do it as well, because that was the main feature of `ghodss/yaml`